src/MerchantCenter/PolicyComplianceCheck.php

```diff
@@ -0,0 +1,230 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\MerchantCenter;
+
+use Automattic\WooCommerce\GoogleListingsAndAds\Infrastructure\Service;
+use Automattic\WooCommerce\GoogleListingsAndAds\Proxies\WC;
+use Automattic\WooCommerce\GoogleListingsAndAds\Google\GoogleHelper;
+
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class PolicyComplianceCheck
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\MerchantCenter
+ *
+ * @since x.x.x
+ */
+class PolicyComplianceCheck implements Service {
+	/**
+	 * The WC proxy object.
+	 *
+	 * @var wc
+	 */
+	protected $wc;
+
+	/**
+	 * @var GoogleHelper
+	 */
+	protected $google_helper;
+
+	/**
+	 * BaseController constructor.
+	 *
+	 * @param WC           $wc
+	 * @param GoogleHelper $google_helper
+	 */
+	public function __construct( WC $wc, GoogleHelper $google_helper ) {
+		$this->wc            = $wc;
+		$this->google_helper = $google_helper;
+	}
+
+	/**
+	 * Check if the store website is accessed by all users for the controller.
+	 *
+	 * @return bool
+	 */
+	public function is_accessible(): bool {
+		$all_countries = $this->wc->get_countries();
+		$mc_countries  = $this->google_helper->get_mc_supported_countries();
+
+		foreach ( $mc_countries as $country ) {
+			if ( ! array_key_exists( $country, $all_countries ) ) {
+				return false;
+			}
+		}
+		return true;
+	}
+
+	/**
+	 * Check if the store sample product landing pages lead to a 404 error.
+	 *
+	 * @return bool
+	 */
+	public function has_page_not_found_error(): bool {
+		$ids = wc_get_products(
+			[
+				'return' => 'ids',
+				'limit'  => 1,
+			]
+		);
+		if ( ! empty( $ids ) ) {
+			$url     = get_permalink( $ids[0] );
```

Comment:
Since we are using `wc_get_products` it would be nice to return the object (remove return parameter) and then use the crud function to [get the permalink](https://woocommerce.github.io/code-reference/classes/WC-Product.html#method_get_permalink). Ex: `$ids[0]->get_permalink()`